source/manual/errors.html.md

```diff
@@ -8,9 +8,33 @@ last_reviewed_on: 2020-05-18
 review_in: 6 months
 ---
 
-## How we categorise errors
+## How we serve errors on GOV.UK
 
-Sometimes applications will encounter exceptions. This policy describes what we should do for different types of errors. It was first proposed in [RFC 87](https://github.com/alphagov/govuk-rfcs/blob/master/rfc-087-dealing-with-errors.md).
+When a request to GOV.UK fails, we need to handle the error in some way, so that GOV.UK does not look completely broken. Rather than embedding this into every app, we have multiple layers of error handling.
+
+1. **CDN**. When the origin servers [are down](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L36), we:
+
+  - Try to [serve the page from a sequence of static mirrors](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L274)
+  - Serve [an inline 404 page](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L483) if the current mirror returns a 404
+  - Serve [a basic inline error page](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L531) if all the mirrors fail (uh oh!)
+
+1. **Origin**. When a request fails in an upstream app, we:
+
+  - [Intercept the response in Nginx](https://github.com/alphagov/govuk-puppet/blob/7dafec7cccd8308ec90c28835de70243d79b323b/modules/router/templates/router_include.conf.erb#L81)
+  - Replace the response with [a pre-rendered one from Static](https://github.com/alphagov/govuk-puppet/blob/7dafec7cccd8308ec90c28835de70243d79b323b/modules/router/manifests/errorpage.pp#L14)
+
+1. **App**. When an app raises an exception, we:
+
+  - Should [catch the error and return an appropriate error code](https://github.com/alphagov/email-alert-frontend/blob/a2bd35b5b17b7da40cd43df9c2756b564597b66e/app/controllers/application_controller.rb#L10)
+  - Return a "500 Internal Server Error" response (by default)
```

Comment:
This reads as contradictory to me. E.g. if an app can't find the resource a user has requested, should it return a 404 or a 500?
(I think line 29 can be omitted, unless I'm missing a nuance).

Reply:
I see what you mean. I've updated this to imply that the behaviour is inconsistent across our apps.